pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/invertedindex/RealtimeLuceneTextIndex.java

```diff
@@ -151,9 +151,9 @@ public MutableRoaringBitmap getDocIds(String searchQuery) {
       return searchFuture.get();
     } catch (InterruptedException e) {
       docIDCollector.markShouldCancel();
-      LOGGER.warn("TEXT_MATCH query timeout on realtime consuming segment {}, column {}, search query {}", _segmentName,
-          _column, searchQuery);
-      throw new RuntimeException("TEXT_MATCH query timeout on realtime consuming segment");
+      LOGGER.warn("TEXT_MATCH query interrupted while querying the consuming segment {}, column {}, search query {}",
```

Comment:
I changed this message to `timeout` due to some concerns in the original PR, but it's also used during early termination - and calling those queries timeouts is misleading.